internal/devbox/packages.go

```diff
@@ -43,6 +43,40 @@ const StateOutOfDateMessage = "Your devbox environment may be out of date. Run %
 // packages.go has functions for adding, removing and getting info about nix
 // packages
 
+type UpdateVersion struct {
+	Current string
+	Latest  string
+}
+
+// Outdated returns a map of package names to their available latest version.
+func (d *Devbox) Outdated(ctx context.Context) (map[string]UpdateVersion, error) {
+	ctx, task := trace.NewTask(ctx, "devboxOutdated")
+	defer task.End()
+
+	lockfile := d.Lockfile()
+	outdatedPackages := map[string]UpdateVersion{}
```

Comment:
nit, would prefer to use a slice over map. We don't need random access or deduplication.

Reply:
@gcurtis may not like this but we could just return `[]lo.Tuple2[*lockfile.Package, *lockfile.Package]`

that way we don't have to create a new struct at all.

Reply:
hm I don't mind the struct. Its nice to have the names

Reply:
How you prefer guys, here map doesn't create any performance problems, it is just a matter of some seconds to run the command, I chose readability over little optimization